java10/scripts.js

```diff
@@ -0,0 +1,145 @@
+var newGameBtn = document.getElementById('js-newGameButton'),
+    newGameElem = document.getElementById('js-newGameElement'),
+    pickElem = document.getElementById('js-playerPickElement'),
+    resultsElem = document.getElementById('js-resultsTableElement');
+
+newGameBtn.addEventListener('click', newGame);
+
+var gameState = 'notStarted',
+    player = {
+        name: '',
+        score: 0
+    },
+    computer = {
+        score: 0
+    };
+
+function setGameElements() {
+    switch(gameState) {
+        case 'started':
+            newGameElem.style.display = 'none';
+            pickElem.style.display = 'block';
+            resultsElem.style.display = 'block';
+        break;
+        case 'ended':
+            newGameBtn.innerText = 'Try again';
+            newGameElem.style.display = 'block';
+            pickElem.style.display = 'none';
+            resultsElem.style.display = 'block';
+        break;
+        case 'notStarted':
+        default:
+            newGameElem.style.display = 'block';
+            pickElem.style.display = 'none';
+            resultsElem.style.display = 'none';
+    }
+}
+
+setGameElements();
+
+var playerPointsElem = document.getElementById('js-playerPoints'),
+    playerNameElem = document.getElementById('js-playerName'),
+    computerPointsElem = document.getElementById('js-computerPoints');
+
+function newGame() {
+    player.name = prompt('Please, enter your name');
+
+    if (player.name) {
+        player.score = computer.score = 0;
+        gameState = 'started';
+        setGameElements();
+
+        playerNameElem.innerHTML = player.name;
+        setGamePoints(); 
+    }
+}
+
+function endGame() {
+    if(player.score == 10) {
+        alert(`Koniec rozgrywki wygral ${player.name} wynikiem: ${player.score} : ${computer.score} `);
+        gameState = 'ended';
+        setGameElements();    
+    } else if(computer.score == 10){
+        alert(`Koniec rozgrywki wygral komputer wynikiem: ${computer.score} : ${player.score}`);
+        gameState = 'ended';
+        setGameElements();    
+    }
+}
+
+var pickRock = document.getElementById('js-playerPick_rock'),
+    pickPaper = document.getElementById('js-playerPick_paper'),
+    pickScissors = document.getElementById('js-playerPick_scissors');
+
+pickRock.addEventListener('click', function() { playerPick('Rock') });
+pickPaper.addEventListener('click', function() { playerPick('Paper') });
+pickScissors.addEventListener('click', function() { playerPick('Scissors') });
+
```

Comment:
to samo, jest to kod leżący 'luźno' pomiędzy deklaracjami - ciężko się stwierdza kiedy te deklaracje mają miejsce. Rejestrację event listenerów można nawet pokusić się ubrać w funkcję i wywoływać ją na początku jako `registerListeners()`